cmake/StaticBuild.cmake

```diff
@@ -23,7 +23,10 @@ if(BUILD_STATIC)
 
   # gettext is included in libc on many unix systems
   if(NOT LIBC_HAS_DGETTEXT)
-    set(GETTEXT_LIBRARIES "-Wl,-Bstatic -lintl -liconv -Wl,-Bdynamic")
+    set(GETTEXT_LIBRARIES "-Wl,-Bstatic -lidn2 -lintl -lunistring -liconv -Wl,-Bdynamic")
```

Comment:
I'm afraid this breaks some setups. Please have a look at #1032 and the discussion there.

Reply:
These are the changes @bphinz mentioned and uses in the official build process.  I offset the other changes based on the MinGW Makefiles generator.

Reply:
There is a suggested fix from me on that other issue. Could you test that?

Reply:
I'll try it out and get back to you.

Reply:
Your approached worked with my build system.  I made the changes.